File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        if (x < 10) {
            return true;
        }

        int sum = 0;
        char[] chars = String.valueOf(x).toCharArray();
        for (char aChar : chars) {
            sum += aChar - 48;
        }

        return x % sum == 0;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    string str = to_string(x);
    int sum = int{0};
    for(int i = 0; i < str.length(); i++) {
        sum += str.at(i) - '0';
    }
    return x % sum == 0;
}